appcues/src/main/java/com/appcues/statemachine/states/RenderingStepState.kt

```diff
@@ -90,18 +90,20 @@ internal data class RenderingStepState(
     }
 
     private fun toEndingExperience(action: EndExperience): Transition {
+        // passing in the effect on EndingStepState to ensure the UI will dismiss
+        // even though we might not be waiting for the completion before moving to
+        // the next step.
+        val awaitDismissEffect = AwaitDismissEffect(action)
         return if (action.destroyed) {
             // this means the activity hosting the experience was destroyed externally (i.e. deep link) and we should
             // immediately transition to EndingExperience - not rely on the UI to do it for us (it's gone)
-            next(EndingStepState(experience, flatStepIndex, action.markComplete, null), ContinuationEffect(action))
+            next(EndingStepState(experience, flatStepIndex, action.markComplete, awaitDismissEffect), ContinuationEffect(action))
```

Comment:
checking understanding - the `EndingStepState.awaitDismissEffect` is still optional, and thus no changes in the `AppcuesViewModel` implementation where it does the null check. However, it should really never be null there, in practice.

The one spot it could still be null is when transitioning in `EndingStepState.toBeginningStepState` where it is checking if a new container will be presented or not.

Reply:
Thats right, so one thing we missed when we last talked is that if we had the uiState dismiss to go through the container dismisses, this is bad for multiple steps (failed snapshot tests), one thing we could do to make it clearer is to create pass in a "isDestroyed" boolean on EndingStepState to do this. decided to keep the State as is but open to change.

Reply:
ok sounds good, and good catch. seems ok to me to move this fix along to resolve the observed issue, thanks!